Sources/SwiftInspectorVisitors/TypeDescription.swift

```diff
@@ -104,6 +105,24 @@ public enum TypeDescription: Codable, Hashable {
       } else {
         return "\(parentType.asSource).\(name)<\(generics.map { $0.asSource }.joined(separator: ", "))>"
       }
+    case let .attributed(type, specifier: specifier, attributes: attributes):
+      func attributesFromList(_ attributes: [String]) -> String {
+        attributes
+          .map { "@\($0)" }
+          .joined(separator: " ")
+      }
+      switch (specifier, attributes) {
+      case let (.some(specifier), .none):
+        return "\(specifier) \(type.asSource)"
+      case let (.none, .some(attributes)):
+        return "\(attributesFromList(attributes)) \(type.asSource)"
+      case let (.some(specifier), .some(attributes)):
+        // This case likely represents an error.
+        return "\(specifier) \(attributesFromList(attributes)) \(type.asSource)"
```

Comment:
I'm reasonably sure this ordering is both correct and stable.

I was playing around with https://swift-ast-explorer.com/ and `inout @autoclosure () -> Void` parsed as an `AttributedType` but `@autoclosure inout () -> Void` parsed into `UnknownType`... so I'm pretty sure the specifier has to come first.

That said, [the Swift reference manual](https://docs.swift.org/swift-book/ReferenceManual/Types.html#grammar_type) shows the specifier (specifically `inout`) coming after attributes:
<img width="406" alt="image" src="https://user-images.githubusercontent.com/139364/140258365-968c5db3-0328-4519-818d-d4088ceec504.png">

Not really sure what to do with this one. Though given that I haven't yet come up with a compiling statement where both the specifier and attributes are present I'm not super worried about it.

If you can come up with a compiling type declaration that has both a specifier and attributes, I'd love to see it.

Reply:
Note that because I believe this is an error I haven't been able to write a test case that covers this case.

Reply:
I wondered if we could do something with property wrappers on function parameters though they don't support `inout` at the moment https://github.com/apple/swift-evolution/blob/main/proposals/0293-extend-property-wrappers-to-function-and-closure-parameters.md#support-inout-in-wrapped-function-parameters

Reply:
Let's add a comment that we are aware this conflicts with the grammar and explain why we are writing this the way we are.

Reply:
Per [my comment in the other thread](https://github.com/fdiaz/SwiftInspector/pull/120/files#r743379615) I don't think property wrappers or function parameters are attributes on the type, but rather are attributes on the property.

I've added a better block comment here however. af09cb1